include/frg/formatting.hpp

```diff
@@ -476,6 +488,151 @@ struct escape_fmt {
 	size_t _size;
 };
 
+namespace detail_ {
+	template <typename ...Ts>
+	struct fmt_impl {
+		frg::string_view fmt;
+		frg::tuple<Ts...> args;
+
+		template <typename F>
+		bool format_nth(size_t n, format_options fo, F &formatter) const {
+			if (n >= sizeof...(Ts))
+				return false;
+
+			return ![&]<size_t ...I>(std::index_sequence<I...>) -> bool {
+				return ((I == n
+					? (format(args.template get<I>(), fo, formatter), false)
+					: true) && ...);
+			}(std::make_index_sequence<sizeof...(Ts)>{});
+		}
+
+		// Format specifier syntax:
+		// ([0-9]+)?(:0?[0-9]*[dXx]?)?
+		bool parse_fmt_spec(frg::string_view spec, size_t &pos, format_options &fo) const {
+			size_t i = 0;
+
+			if (i < spec.size() && isdigit(spec[i])) {
+				pos = 0;
+				i++;
+
+				while(i < spec.size() && isdigit(spec[i])) {
+					pos *= 10;
+					pos += spec[i] - '0';
+					i++;
+				}
+			}
+
+			// Specifying only the argument index is fine
+			if (i == spec.size())
```

Comment:
Why can't we just use a for loop + a small state machine to parse this? Would avoid having to use all these checks.